packages/tests/test/reya_common/trade/PassivePool.fork.c.sol

```diff
@@ -255,4 +266,288 @@ contract PassivePoolForkCheck is BaseReyaForkTest {
+        }
+
+        for (uint256 i = 0; i < allTokens.length; i++) {
+            for (uint256 j = 0; j < allTokens.length; j++) {
+                if (i == j) {
+                    continue;
+                }
+
+                address tokenIn = allTokens[i];
+                address tokenOut = allTokens[j];
+
+                uint256 amountIn = 100_000_000 * (10 ** IERC20TokenModule(tokenIn).decimals());
+
+                RebalanceAmounts memory rebalanceAmounts =
+                    IPassivePoolProxy(sec.pool).getRebalanceAmounts(sec.passivePoolId, tokenIn, tokenOut, amountIn);
+
+                if (rebalanceAmounts.amountIn != 0) {
+                    vm.prank(dec.socketController[tokenIn]);
+                    IERC20TokenModule(tokenIn).mint(sec.rebalancer1, rebalanceAmounts.amountIn);
+
+                    vm.prank(sec.rebalancer1);
+                    IERC20TokenModule(tokenIn).approve(sec.pool, rebalanceAmounts.amountIn);
+
+                    vm.prank(sec.rebalancer1);
+                    IPassivePoolProxy(sec.pool).triggerAutoRebalance(
+                        sec.passivePoolId,
+                        AutoRebalanceInput({
+                            tokenIn: tokenIn,
+                            amountIn: rebalanceAmounts.amountIn,
+                            tokenOut: tokenOut,
+                            minPrice: rebalanceAmounts.priceInToOut,
+                            receiverAddress: sec.periphery
+                        })
+                    );
+
+                    vm.warp(block.timestamp + 1);
+                }
+            }
+        }
+    }
+
+    function check_autoRebalance_currentTargets() public {
+        autoRebalancePool();
+    }
+
+    function check_autoRebalance_differentTargets() public {
+        vm.prank(sec.multisig);
+        IPassivePoolProxy(sec.pool).setAllocationConfiguration(
+            sec.passivePoolId, AllocationConfigurationData({ quoteTokenTargetRatio: 0.353535e18 })
+        );
+
+        address[] memory supportingCollaterals =
+            IPassivePoolProxy(sec.pool).getQuoteSupportingCollaterals(sec.passivePoolId);
+        uint256[] memory newSupportingCollateralsAllocations = new uint256[](supportingCollaterals.length);
+
+        newSupportingCollateralsAllocations[newSupportingCollateralsAllocations.length - 2] = 0.454545e18;
+        newSupportingCollateralsAllocations[newSupportingCollateralsAllocations.length - 1] = 1e18 - 0.454545e18;
+
+        vm.prank(sec.multisig);
+        IPassivePoolProxy(sec.pool).setAllocations(sec.passivePoolId, newSupportingCollateralsAllocations);
+
+        autoRebalancePool();
+    }
+
+    function check_autoRebalance_noSharePriceChange() public {
+        uint256 sharePrice0 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+        check_autoRebalance_differentTargets();
+        uint256 sharePrice1 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+        assertLe(sharePrice0, sharePrice1);
```

Comment:
hmm, it might be worth doing a few more different rebalancing variations, e.g. in some variations rusd is in deficit and deusd is above target or rusd is at target and we just need to swap deusd for sdeusd, etc (it's effectively then a matrix of pre-conditions or us to go through)